synapse/_scripts/synapse_port_db.py

```diff
@@ -1307,7 +1307,7 @@ def main() -> None:
     sqlite_config = {
         "name": "sqlite3",
         "args": {
-            "database": args.sqlite_database,
+            "database": "file:{}?mode=rw".format(args.sqlite_database),
```

Comment:
I'm not familiar with this syntax -- how is this ensuring the file exists? What sort of error does it provide if the file does *not* exist?

(Also -- does the porter need write access to the sqlite file?)

Reply:
Currently when sqlite db file doesn't exist it creates an empty file as by default the file is accessed in rwc (read-write-create) mode.
The fix here switches it to open in rw mode, it would ensure the file is not created if it doesn't exist. It still has write access to the file.
This is the error returned back when there is no sqlite DB file:
![image](https://user-images.githubusercontent.com/6557230/208926439-fe937196-05c3-4fd1-aedb-45b73a764b03.png)

Reply:
Right. This is from SQLite's [URI syntax](https://www.sqlite.org/uri.html).

Reply:
> The fix here switches it to open in rw mode, it would ensure the file is not created if it doesn't exist. It still has write access to the file.

Do we need write access though? Doesn't the porter only care about reading data?

Reply:
That makes sense, I will change it to read only.

Reply:
@JeyRathnam Ah, sorry for the bad guidance. It seems we *do* need to write also since it upgrades the sqlite database before porting it. (I guess this makes sense? I'm not 100% sure it is on purpose.)

Reply:
Pushed a fix to change to read-write. Do you think I should write a specific test case for this? I ran tests locally when I switched to read-only and all tests passed.

Reply:
@JeyRathnam There's a separate port DB test in CI that picked this up, see the failed job @ https://github.com/matrix-org/synapse/actions/runs/3753491620/jobs/6376837014

I don't think we need to write a separate test in addition to that.